py/dml/output.py

```diff
@@ -64,6 +64,12 @@ def reset_line_directive(self):
             self.out('#line %d "%s"\n'
                      % (self.lineno + 1, quote_filename(self.filename)))
 
+    class IndeterminateFilename: pass
```

Comment:
I don't understand -- no code checks for this, so what's the desired property? that it's `!=` all paths and `None`? Is it ever `__str__`ed?

Consider singletoning the class, to mark the intent of an object without properties, as opposed to a class without properties (I sometimes use a decorator `@lambda cls: cls()`; I miss the `@apply` construct from py2).